### ISSUE TITLE
Performance decrease when there is a lot of validation stamps.

### ISSUE THREAD

**Issue by flesire** [nemerosa/ontrack#372]
We have projects with branches that contain approximately 40 validation stamps. With these projects the rendering of the branch page take 1minute 40 seconds to display the build progress indicator. If i filter to reduce the number of displayed validation stamps to 5 columns, the performance is good.

Remark : I use the latest version of ontrack (2.17.4)

**Comment by dcoraboeuf** [comment]
What do you mean by build progress indicator?

**Comment by flesire** [comment]
I mean the one that appears before all the builds are displayed.

**Comment by dcoraboeuf** [comment]
Hi Fabien,

If you look at the Network tab in the Developer Tools in Chrome for example, and if you reload the page, can you tell me which XHR request(s) take(s) the most time in your case? And send me the associated JSON (after being obfuscated as needed of course).

Thanks,
Damien.

**Comment by flesire** [comment]
![image](https://cloud.githubusercontent.com/assets/1262565/12192785/08afe614-b5e2-11e5-944b-8257460d9ace.png)

All the requests are validations stamp for weather status of jenkins jobs. Here is one of the json :
```json
{
	"_self": "http://.../decorations/VALIDATION_STAMP/1621",
	"resources": [{
		"id": "sunny",
		"title": "Sunny (0 failure in the last 5 builds)",
		"uri": null,
		"name": null,
		"decorationType": "net.nemerosa.ontrack.extension.general.ValidationStampWeatherDecorationExtension"
	},
	{
		"id": "idle",
		"title": "The Jenkins Job is not running.",
		"uri": null,
		"name": null,
		"decorationType": "net.nemerosa.ontrack.extension.jenkins.JenkinsJobDecorationExtension"
	}]
}
```

Remark : I've tested the same scenario with latest release and same problem.

**Comment by dcoraboeuf** [comment]
The collection of decorations is asynchronous - performed after the builds and the validation stamps have been loaded. Their collection definitely needs optimisation but that should not explain your current situation. Can you tell me the duration & content size for the `structure/branches/186/view` call?

**Comment by flesire** [comment]
Duration : 1.66s
Content-Length:1178057

Json response : [view_json.txt](https://github.com/nemerosa/ontrack/files/82454/view_json.txt)

**Comment by dcoraboeuf** [comment]
Issues #397 and #399 should already improve the performances quite a lot. They are available right now on the `master` branch and will be released with 2.20. I'll keep working on reducing the load on the client side.

**Comment by dcoraboeuf** [comment]
See also #405

**Comment by dcoraboeuf** [comment]
Performances have been increased a lot in the home, project and branch pages.

**Comment by dcoraboeuf** [comment]
Will be available in 2.20 and already in the `master` branch.
